osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/mapping/customconverters/EventsConverter.java

```diff
@@ -33,8 +35,10 @@ public EventDto convertFrom(final Event source, final Type<EventDto> destination
         if (source == null) {
             return null;
         }
-
-        return new EventDto(source.getTimestamp(), source.getEventCode(), source.getEventCounter(),
+        final EventDto eventDto = new EventDto(source.getTimestamp(), source.getEventCode(),
+                source.getEventCounter(),
                 source.getEventLogCategory().name());
+        eventDto.setEventTypeDto(EventTypeDto.valueOf(source.getEventType().name()));
```

Comment:
can be added to constructor ?

Reply:
This is not possible, cause we add the EventType afterwards in the EventService, based on the eventCode